src/main/java/uk/gov/mca/beacons/service/dto/HateoasLinkBuilder.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.mca.beacons.service.dto;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder;
+import org.springframework.stereotype.Service;
+import uk.gov.mca.beacons.service.beacons.BeaconsController;
+import uk.gov.mca.beacons.service.model.Beacon;
+
+@Service
+public class HateoasLinkBuilder {
+
+  public enum SupportedMethod {
+    GET,
+    PATCH,
+  }
+
+  private final Class<BeaconsController> beaconController =
+    BeaconsController.class;
+
+  public void addLinkFor(Beacon domain, SupportedMethod method, BeaconDTO dto) {
```

Comment:
As discussed, might be worth looking at removing the context-path of `/spring-api` as this is prefixed to all API routes.

Will require updating a few environment variables in the integration project:

- https://github.com/mcagov/beacons-integration/blob/main/terraform/variables.tf#L66
- https://github.com/mcagov/beacons-integration/blob/main/terraform/provider.tf#L27